### DISCUSSION TITLE
Watching files outside of project directory

### DISCUSSION THREAD

**Discussion by Flamenco** [quasarframework/quasar#17389] [CLI - BEX mode]
Is there a way to watch files outside of the project directory when building BEX project.

I understand that refreshing the browser is needed.  My is issue is that I use a library on my machine outside the project directory, and the filewatcher will not rebuild when its source files change.  I need to kill and restart the project every time a file changes, then reload the browser after 10-20 seconds.

I am using Vite, and there is no way I can see in the config to add additional paths either.

**Comment by stefanvanherwijnen** [comment]
https://github.com/vitejs/vite/issues/14423#issuecomment-1728905288

**Comment by Flamenco** [comment]
I found a workaround for this, as after research it seems the required functionality is not available via `Vite` or in turn `chokidar`, due to the way Quasar wraps those libraries.

Install nodemon
```
yarn add -D nodemon
```
Then start `Quasar` with `nodemon`, making sure to add the desired watch directories and files, and also add desired extensions (`vue` files are not watched by default)

```json
"scripts": {
  "dev": "nodemon --watch ../dir1 --watch ../dir2 --ext vue,js,ts --exec quasar dev -m bex"
}
```

You MUST either add `--watch` directories, or `--exclude ./` to prevent an end endless restart cycle.

This is not a hot-reload for the watch directories, but is still hot-reloaded for the project directory, and is much better than `ctrl-c` over and over.
